.src/state.cpp

```diff
@@ -42,7 +42,28 @@ namespace jluna::State
         else
             jl_init_with_image(path.c_str(), NULL);
 
-        jl_eval_string(jluna::detail::include);
+        const char* julia_lib_load_template = R"julia(
+          begin
+            local dev_path = "%s/include/jluna.jl"
+            local prod_path = "%s/jluna/jluna.jl"
+            if isfile(dev_path)
+              include(dev_path)
+            else isfile(prod_path)
+              include(prod_path)
+            end
+          end
+        )julia";
+        std::vector<char> buf (
+            /* no need to account for null since %s takes 4 bytes accounted for
+             * which is double counted by BUILD_RESOURCE_DIR and RUN_RESOURCE_DIR */ 
+            strlen(julia_lib_load_template) +
+            strlen(BUILD_RESOURCE_DIR) +
+            strlen(RUN_RESOURCE_DIR),
+            /*c strings are null terminated*/
+            '\0'
+        );
+        snprintf(buf.data(), buf.size(), julia_lib_load_template, BUILD_RESOURCE_DIR, RUN_RESOURCE_DIR);
+        jl_eval_string(buf.data());
```

Comment:
While std::format would be nice, I feel like assembling this with std::stringstream and then calling `jl_eval_string(stringstream_var.str().c_str()` would've been a lot easier, but fair play to you for doing it C-style

Reply:
Also how come std::format is the single feature where clang is further than gcc, very odd, usually gcc is much faster at implementing the new stuff.